File: rust/src/trae.rs
pub struct TraeAgent {
    pub builder_mode: bool,
}

impl TraeAgent {
    pub fn new() -> Self {
        Self { builder_mode: false }
    }

    pub fn set_builder_mode(&mut self, enabled: bool) {
        self.builder_mode = enabled;
        println!("Trae Builder Mode set to: {}", enabled);
    }
}

File: go/internal/agent/trae.go
package agent

import (
	"context"
	"fmt"
)

type TraeAgent struct {
	BaseAgent
	BuilderMode bool
}

func NewTraeAgent() *TraeAgent {
	return &TraeAgent{BuilderMode: false}
}

func (a *TraeAgent) SetBuilderMode(enabled bool) {
	a.BuilderMode = enabled
	fmt.Printf("Trae Builder Mode set to: %v\n", enabled)
}
